src/main/java/lotterymachine/LotteryMachine.java

```diff
@@ -41,30 +46,71 @@ private static Money getInputAmount() {
     }
 
     private static Count getNumberOfTickets(Money money) {
-        Count numberOfTickets = Count.from(money.divideByTicketPrice());
-        OutputView.printNumberOfTicket(numberOfTickets.getNumber());
-        return numberOfTickets;
+        return Count.from(money.divideByTicketPrice());
     }
 
-    private static LotteryTickets purchaseLotteryTickets(Count numberOfTickets) {
-        List<LotteryTicket> lotteryTicketsList = generate(numberOfTickets);
-        LotteryTickets lotteryTickets = new LotteryTickets(lotteryTicketsList);
+    private static Count getNumberOfManualTickets(Count totalTickets) {
+        try {
+            int numberOfManualTickets = InputView.getNumberOfManualTickets();
+            return Count.of(totalTickets, numberOfManualTickets);
+        } catch (RuntimeException exception) {
+            OutputView.printException(exception.getMessage());
+            return getNumberOfManualTickets(totalTickets);
+        }
+    }
+
+    private static LotteryTickets purchaseLotteryTickets(Count numberOfTickets, Count manualTickets) {
+        List<LotteryTicket> manualLotteryTickets = purchaseManualLotteryTickets(manualTickets);
+        Count autoTickets = numberOfTickets.subtract(manualTickets);
+        List<LotteryTicket> autoLotteryTickets = generate(autoTickets);
+        LotteryTickets lotteryTickets = collectTickets(manualLotteryTickets, autoLotteryTickets);
+
+        OutputView.printPurchaseDetails(manualTickets.getNumber(), autoTickets.getNumber());
         OutputView.printLotteryTickets(lotteryTickets.getLotteryTickets());
         return lotteryTickets;
     }
 
+    private static LotteryTickets collectTickets(List<LotteryTicket> manualTickets, List<LotteryTicket> autoTickets) {
+        return new LotteryTickets(Stream.of(manualTickets, autoTickets)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toUnmodifiableList()));
+    }
+
+    private static List<LotteryTicket> purchaseManualLotteryTickets(Count manualTickets) {
+        List<LotteryTicket> lotteryTickets = new ArrayList<>();
+        if (!manualTickets.isInteger()) {
+            return lotteryTickets;
+        }
+        OutputView.printInputManualPurchase();
+        for (int i = 0; i < manualTickets.getNumber(); i++) {
+            lotteryTickets.add(getManualLotteryTicket());
+        }
+        return lotteryTickets;
```

Comment:
```suggestion
    private static List<LotteryTicket> purchaseManualLotteryTickets(Count manualTickets) {
        List<LotteryTicket> lotteryTickets = new ArrayList<>();
        OutputView.printInputManualPurchase();
        for (int i = 0; i < manualTickets.getNumber(); i++) {
            lotteryTickets.add(getManualLotteryTicket());
        }
        return lotteryTickets;
```
컨트롤러에 비즈니스를 최대한 담지 않도록 해보는게 어떨까요?
만약 0개를 입력했다면 `OutputView.printInputManualPurchase(manualTickets.isZero());` 이런식으로 출력문을 조정해볼 수 있을 것 같아요.

Reply:
추가적으로 사라질 코드지만,
```
if (!manualTickets.isInteger()) {
            return lotteryTickets;
        }
```
- 음수같은 다른값도 포함해서 빈리스트로 반환하려는게 보였는데요. 이런부분은 대체값을 사용하기보다는 예외를 발생해서 빠르게 피드백하는게, 추후 디버깅하기나 흐름이해가 더 편할 수 있어요.
- if문을 작성할 때 부정(!)이 들어간다면 메서드 시그니쳐를 바꾸는게 읽기 더 편할 것 같아요. `manualTickets.isNotInteger()`

Reply:
감사합니다! 해당 피드백 반영해보았습니다! 😊😊